agent/github_tools.py

```
@@ -1,658 +1,658 @@
+@github_router.get("/repos")
+async def list_repos(request: Request):
+    """List GitHub repos accessible to the current user."""
+    user  = _get_user(request)
+    token = await _get_token(user)
+    gh    = GitHubTools(token=token)
+    try:
+        repos = await gh.list_repos()
+        return {"repos": repos, "total": len(repos)}
+    except httpx.HTTPStatusError as e:
+        raise HTTPException(status_code=e.response.status_code,
+                            detail=f"GitHub API error: {e.response.text}")
```

Ruff (0.15.18)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)